apps/blockchain/lib/blockchain/block.ex

```diff
@@ -906,4 +932,25 @@ defmodule Blockchain.Block do
 
     %{block | block_hash: hash, header: updated_block}
   end
+
+  defp store_transaction_locations_by_hash(trie, block) do
+    block.transactions
+    |> Enum.with_index()
+    |> Enum.reduce(trie, fn {transaction, idx}, trie_acc ->
+      transaction_key =
+        transaction
+        |> Transaction.hash()
+        |> transaction_key()
+
+      location = :erlang.term_to_binary({block.block_hash, idx})
+
+      TrieStorage.put_raw_key!(trie_acc, transaction_key, location)
+    end)
+  end
+
+  defp transaction_key(hash) do
+    hash_hex = Base.encode16(hash, case: :lower)
+
+    "transaction_#{hash_hex}"
```

Comment:
The hash should be unique, right? Probably don't need "transaction_", but it doesn't hurt much.